frappe/integrations/doctype/token_cache/token_cache.py

```diff
@@ -52,8 +52,8 @@ def update_data(self, data):
 		return self
 
 	def get_expires_in(self):
-		expiry_time = frappe.utils.get_datetime(self.modified) + timedelta(self.expires_in)
-		return (datetime.now() - expiry_time).total_seconds()
+		expiry_time = frappe.utils.get_datetime(self.modified) + timedelta(seconds=self.expires_in)
+		return (expiry_time - frappe.utils.now_datetime()).total_seconds()
```

Comment:
Isn't it `return (frappe.utils.now_datetime() - expiry_time).total_seconds() `

Reply:
NO
If you look at the calculation above, expiry time is in the FUTURE. Therefore subtracting now() from a future time will be positive and it will go negative when expired.

You do realise people test their changes before submitting a PR?

Reply:
Thanks for clarifying the silly question (I have confused with the context switch).

It is always good to answer softly if some one has some questions.